### PR TITLE
chore: move onto stable Rust

### LINKED ISSUE #809: chore: switch to stable rust
Ideally we should target stable rust. Currently this is what is preventing us form building on stable rust:

- [ ] arrow rust upstream uses deprecated `packed_simd_2` (which requires nightly). We can either wait or contribute upstream to switch stdsimd which is not deprecated.

### PR DESCRIPTION
**Pull request by e-dard**
Closes #809

Well, some of us are hankering for stable Rust and we have been unable to move onto it because the SIMD feature in our Arrow dependency requires us to tie ourselves to a blessed nightly version. 

If we disable the SIMD feature in Arrow then we can move onto stable Rust. We don't anticipate right now that SIMD optimisations in Arrow are a bottleneck for our performance endeavours so we can disable it for now and revisit when the story changes.

Further, the Read Buffer has some SIMD optimisations that currently don't rely on a nightly version of Rust; they will still work fine.

### CODE REVIEW

**Comment by e-dard** [comment]
I have the test suite locally on an `intel` machine with `avx2` enabled (to execute the SIMD paths in the Read Buffer, as well as `aarch64` and all seems well.

**Comment by e-dard** [comment]
```
#!/bin/bash -eo pipefail
cargo clippy --all-targets --workspace -- -D warnings

info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: latest update on 2021-03-25, rust version 1.51.0 (2fd73fabe 2021-03-23)
info: downloading component 'cargo'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: installing component 'cargo'
info: using up to 500.0 MiB of RAM to unpack components
info: installing component 'rust-std'
info: installing component 'rustc'
error: 'cargo-clippy' is not installed for the toolchain 'stable-x86_64-unknown-linux-gnu'
To install, run `rustup component add clippy --toolchain stable-x86_64-unknown-linux-gnu`

Exited with code exit status 1

CircleCI received exit code 1```

😢

**Review by crepererum — APPROVED** [review]

**Review by domodwyer — APPROVED** [review]
I <3 stable

We'll probably have to make sure stable is installed in the [CI image](https://github.com/influxdata/influxdb_iox/blob/d6c71bae4fe0fd8af606c6643d7bd461c5d747d1/.circleci/config.yml#L28-L36) - should work for now for testing though? I think it'll automatically install stable for each CI step when it sees the toolchain file, but it'll take a while.

File: server/src/config.rs [hunk 1]
```diff
@@ -208,14 +208,14 @@ pub fn object_store_path_for_database_config<P: ObjectStorePath>(
 }
 
 /// A gRPC connection string.
-pub type GRPCConnectionString = String;
+pub type GRpcConnectionString = String;
```
Comment (tustvold) [inline server/src/config.rs:211]: Why is the R still capitalised?

**Pull request merged**